tests/csm/rest/test_information.py

```diff
@@ -0,0 +1,211 @@
+        cls.rest_resp_conf = configmanager.get_config_wrapper(
+            fpath="config/csm/rest_response_data.yaml")
+        cls.log.info("Ended test module setups")
+        cls.config = CSMConfigsCheck()
+        setup_ready = cls.config.check_predefined_s3account_present()
+        if not setup_ready:
+            setup_ready = cls.config.setup_csm_s3()
+        assert setup_ready
+        cls.created_iam_users = set()
+        cls.csm_obj = csm_api_factory("rest")
+        cls.log.info("Initiating Rest Client ...")
+
+    def teardown_method(self):
+        """
+        Teardown method which run after each function.
+        """
+        self.log.info("Teardown started")
+        self.log.info("Teardown ended")
+
+    @pytest.mark.lc
+    @pytest.mark.parallel
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-42713')
+    def test_42713(self):
+        """
+        Test that user can check version compatability.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("[START] Testing Version Compatability with compatible versions")
+        # For Happy path with valid inputs
+        # call api compatible version
+        payload = self.csm_obj.get_version_compatibility_payload("compatible")
+        self.log.info("payload :  %s", payload)
+        response = self.csm_obj.verify_version_compatibility("node", "cortx-control", payload)
+        res_dict = response.json()
+        assert response.status_code == HTTPStatus.OK, "Status code check failed"
+        assert res_dict["compatible"], "Compatibility Check failed"
+        success_reason_msg = "Versions are Compatible"
+        assert res_dict["reason"] == success_reason_msg, "response reason is not correct"
+        self.log.info("[END] Testing Version Compatability")
+
+        self.log.info("##### Test ended -  %s #####", test_case_name)
+
+    @pytest.mark.lc
+    @pytest.mark.parallel
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-42789')
+    def test_42789(self):
+        """
+        Test that user can check version compatability.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+
+        self.log.info("[START] Testing Version Compatability with incompatible version rules")
+        payload = self.csm_obj.get_version_compatibility_payload("incompatible")
+        self.log.info("payload :  %s", payload)
+        response = self.csm_obj.verify_version_compatibility("node", "cortx-control", payload)
+        res_dict = response.json()
+        assert response.status_code == HTTPStatus.OK, "Status code check failed"
+        assert not res_dict["compatible"], "Compatibility Check failed"
+        success_reason_msg = "Versions are Compatible"
+        assert res_dict["reason"] != success_reason_msg, "response reason is not correct"
```

Comment:
As messages are subject to frequent change, can u move message check under below if conditions. ETA for all other tests.
if self.config["msg_check"] == "enable":

Reply:
updated